x-pack/platform/plugins/shared/cases/server/routes/api/internal/find_cases_containing_all_alerts.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { type IKibanaResponse } from '@kbn/core/server';
+import pLimit from 'p-limit';
+import { createCasesRoute } from '../create_cases_route';
+import { INTERNAL_CASE_GET_CASES_BY_ATTACHMENT_URL } from '../../../../common/constants';
+import { DEFAULT_CASES_ROUTE_SECURITY } from '../constants';
+import { type CasesClient } from '../../../client';
+import { type caseApiV1 } from '../../../../common/types/api';
+
+// cases modal shows 10 cases by default
+const MAX_CONCURRENT_CASES = 10;
+
+export const findCasesContainingAllAlertsRoute = createCasesRoute({
+  method: 'post',
+  path: INTERNAL_CASE_GET_CASES_BY_ATTACHMENT_URL,
+  security: DEFAULT_CASES_ROUTE_SECURITY,
+  params: {
+    body: schema.object({
+      alertIds: schema.maybe(schema.oneOf([schema.string(), schema.arrayOf(schema.string())])),
+      caseIds: schema.maybe(schema.oneOf([schema.string(), schema.arrayOf(schema.string())])),
+    }),
```

Comment:
Is there a scenario where either `alertIds` or `caseIds` is not defined or if they're strings?
I'm asking because we can make the body parser quite strict and then won't need to add defensive code further down in the route handler.

For example this one can be removed then because the param parser takes care of that work for us:

```ts
if (!isStringOrArray(caseIds) || !isStringOrArray(alertIds)) {
      return response.badRequest({
        body: { message: 'Invalid request parameters' },
      });
    }
```